pkg/metrics/builder/builder.go

```diff
@@ -0,0 +1,123 @@
+package builder
+
+import (
+	"errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// MetricBuilder is the basic metric type that must be used by any potential metric added to
+// the system. The build function will use the properties of this struct and create a collector
+// which will be used to push.
+type MetricBuilder struct {
+	// labels contains the list of all label keys that the collector object will have.
+	labels []string
+	// labelKeyValues stores all the label keys and values collected from the installer and
+	// will be used to create the collector object with the stored values.
```

Comment:
```suggestion
	// labelKeyValues stores values for label keys.
```

Reply:
Also i'm not clear how this is different from just Value. maybe provide a little more info?